src/main/python/ayab/plugins/ayab_plugin/ayab_control.py

```diff
@@ -401,28 +386,28 @@ def __cnfLine(self, lineNumber):
                              or self.__knitting_mode
                              is KnittingMode.HEARTOFPLUTO_RIBBER.value)):
 
-                for col in range(0, 200):
+                for col in range(0, MACHINE_WIDTH):
                     if col < imgStartNeedle \
                             or col > imgStopNeedle:
-                        self._set_pixel(bytes, col)
+                        bits[col] = True
 
             for col in range(0, self.__image.imgWidth()):
                 pxl = (self.__image.imageExpanded())[indexToSend][col]
                 # take the image offset into account
                 if pxl == 1 and sendBlankLine is False:
                     pxlNumber = col + self.__image.imgStartNeedle()
                     # TODO implement for generic machine width
-                    if 0 <= pxlNumber and pxlNumber < 200:
-                        self._set_pixel(bytes, pxlNumber)
+                    if 0 <= pxlNumber and pxlNumber < MACHINE_WIDTH:
+                        bits[pxlNumber] = True
 
             # TODO implement CRC8
             crc8 = 0x00
 
             # send line to machine
             if self.__infRepeat:
-                self.__ayabCom.cnf_line(reqestedLine, bytes, 0, crc8)
+                self.__ayabCom.cnf_line(reqestedLine, bits.tobytes(), 0, crc8)
```

Comment:
Have you verified that the bits.tobytes() operation gives expected result (i.e. have you knitted successfully?)

Reply:
Only on toy examples so far, but I'll get round to it.

Reply:
It needed a change to ensure an integer index. The ordering of bits is correct.